src/main/java/me/jellysquid/mods/lithium/mixin/alloc/vectors/EntityTrackerEntryMixin.java

```diff
@@ -0,0 +1,72 @@
+package me.jellysquid.mods.lithium.mixin.alloc.vectors;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.server.network.EntityTrackerEntry;
+import net.minecraft.util.math.Vec3d;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+/**
+ * Avoid unnecessary {@link Vec3d} allocations.
+ */
+@Mixin(EntityTrackerEntry.class)
+public class EntityTrackerEntryMixin {
+
+    @Unique
+    private double lithium$x;
+    @Unique
+    private double lithium$y;
+    @Unique
+    private double lithium$z;
+
+    @Redirect(
+            method = "tick",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/util/math/Vec3d;subtract(Lnet/minecraft/util/math/Vec3d;)Lnet/minecraft/util/math/Vec3d;"
+            )
+    )
+    private Vec3d removeAllocation(Vec3d playerPos, Vec3d playerLastPos) {
+        // [VanillaCopy] Magic numbers from EntityS2CPacket.decodePacketCoordinates(ddd)
+        this.lithium$x = this.entity.getPos().x - (this.lastX * 2.44140625E-4D);
```

Comment:
Do you have any information on whether saving 1 vec3d allocating per tick is better than adding 3 double fields?

Reply:
I completely forgot that field access is more expensive than local variables. Removed.